src/tree.rs

```diff
@@ -621,6 +632,151 @@ async fn compress_file(path: impl AsRef<str>, offset: u64, size: Option<u64>) ->
+            .await
+            .map(|mut files| { files.sort_by_key(|entry| std::cmp::Reverse(entry.file_name())); files })
+            .inspect_err(|err| {error!("Cannot read journal dir entries: {err}"); })
+            .unwrap_or_default();
+
+        Box::pin(futures::stream::iter(log_files)
+            .map(|file_entry| file_entry.path())
+            .then(|file_path|
+                async move {
+                    match tokio::fs::File::open(&file_path).await {
+                        Ok(file) => {
+                            let reader = JournalReaderLog2::new(BufReader::new(file.compat()));
+                            let (_, latest_entries) = reader
+                                .filter_map(async |res| res.ok())
+                                .fold((None, Vec::new()), async |(latest_ts, mut latest_entries), entry| {
+                                    match latest_ts {
+                                        None => (Some(entry.epoch_msec), vec![entry]),
+                                        Some(ts) if entry.epoch_msec > ts =>
+                                            (Some(entry.epoch_msec), vec![entry]),
+                                        Some(ts) if entry.epoch_msec == ts => {
+                                            latest_entries.push(entry);
+                                            (Some(ts), latest_entries)
+                                        }
+                                        Some(ts) => (Some(ts), latest_entries),
+                                    }
+                                })
+                                .await;
+                            (!latest_entries.is_empty()).then_some(latest_entries)
+                        }
+                        Err(err) => {
+                            error!("Cannot open file {file_path} while getting the last journal entries in pushLog handler: {err}",
+                                file_path = file_path.to_string_lossy()
+                            );
+                            None
+                        }
+                    }
+                })
+            .filter_map(async |latest_entries| latest_entries))
+            .next()
+            .await
+            .unwrap_or_default()
+    };
+    let local_latest_entry_msec = local_latest_entries.first().map_or(0, |entry| entry.epoch_msec);
+
+    let journal_file = match tokio::fs::OpenOptions::new()
+        .write(true)
+        .append(true)
+        .create(true)
+        .open(&journal_file_path)
+        .await {
+            Ok(file) => file,
+            Err(err) => {
+                error!("Cannot open journal file {} for writing: {}", journal_file_path.to_string_lossy(), err);
+                // Return Ok to the device assuming that the devices are not smart enough to handle
+                // the error in this case. It is not their fault that we cannot create the file.
+                // FIXME: if there is a better approach
```

Comment:
Unfortunately, AFAIK, the devices expect this everytime, returning anything might break them. Also, they do NOT ignore the return value, they at least check that "msg" is "success". I don't know if they are able to handle other return values.

Reply:
Ok, good to know